mm2src/mm2_main/tests/docker_tests/swap_watcher_tests.rs

```diff
@@ -0,0 +1,954 @@
+        .into_inner();
+    log!("error: {:?}", error);
+    match error {
+        ValidatePaymentError::WrongPaymentTx(err) => {
+            assert!(err.contains(INVALID_RECEIVER_ERR_LOG))
+        },
+        _ => panic!(
+            "Expected `WrongPaymentTx` tx output script_pubkey doesn't match expected, found {:?}",
+            error
+        ),
+    }
+}
+
+#[test]
+fn test_watcher_spends_maker_payment_spend_utxo() {
+    let (_ctx, _, bob_priv_key) = generate_utxo_coin_with_random_privkey("MYCOIN", 100.into());
+    generate_utxo_coin_with_privkey("MYCOIN1", 100.into(), bob_priv_key);
+    let (_ctx, _, alice_priv_key) = generate_utxo_coin_with_random_privkey("MYCOIN1", 100.into());
+    generate_utxo_coin_with_privkey("MYCOIN", 100.into(), alice_priv_key);
+
+    let watcher_priv_key = random_secp256k1_secret();
+
+    let coins = json!([mycoin_conf(1000), mycoin1_conf(1000)]);
+
+    let alice_conf = Mm2TestConf::seednode_using_watchers(&format!("0x{}", hex::encode(alice_priv_key)), &coins).conf;
+    let mut mm_alice = MarketMakerIt::start(alice_conf.clone(), "pass".to_string(), None).unwrap();
+    let (_alice_dump_log, _alice_dump_dashboard) = mm_dump(&mm_alice.log_path);
+
+    let bob_conf = Mm2TestConf::light_node(&format!("0x{}", hex::encode(bob_priv_key)), &coins, &[&mm_alice
+        .ip
+        .to_string()])
+    .conf;
+    let mm_bob = MarketMakerIt::start(bob_conf, "pass".to_string(), None).unwrap();
+    let (_bob_dump_log, _bob_dump_dashboard) = mm_dump(&mm_bob.log_path);
+
+    let watcher_conf = Mm2TestConf::watcher_light_node(
+        &format!("0x{}", hex::encode(watcher_priv_key)),
+        &coins,
+        &[&mm_alice.ip.to_string()],
+        WatcherConf {
+            wait_taker_payment: 0.,
+            wait_maker_payment_spend_factor: 0.,
+            refund_start_factor: 1.5,
+            search_interval: 1.0,
+        },
+    )
+    .conf;
+    let mut mm_watcher = MarketMakerIt::start(watcher_conf, "pass".to_string(), None).unwrap();
+    let (_watcher_dump_log, _watcher_dump_dashboard) = mm_dump(&mm_watcher.log_path);
+
+    log!("{:?}", block_on(enable_native(&mm_bob, "MYCOIN", &[])));
+    log!("{:?}", block_on(enable_native(&mm_bob, "MYCOIN1", &[])));
+    log!("{:?}", block_on(enable_native(&mm_alice, "MYCOIN", &[])));
+    log!("{:?}", block_on(enable_native(&mm_alice, "MYCOIN1", &[])));
+    log!("{:?}", block_on(enable_native(&mm_watcher, "MYCOIN", &[])));
+    log!("{:?}", block_on(enable_native(&mm_watcher, "MYCOIN1", &[])));
+
+    let rc = block_on(mm_bob.rpc(&json!({
+    "userpass": mm_bob.userpass,
+    "method": "setprice",
+    "base": "MYCOIN",
+    "rel": "MYCOIN1",
+    "price": 25,
+    "max": true,
+    })))
+    .unwrap();
+    assert!(rc.0.is_success(), "!setprice: {}", rc.1);
+
+    let rc = block_on(mm_alice.rpc(&json!({
+    "userpass": mm_alice.userpass,
+    "method": "buy",
+    "base": "MYCOIN",
+    "rel": "MYCOIN1",
+    "price": 25,
+    "volume": "2",
+    })))
```

Comment:
Please indent json content with one tab. There are more spots like this in this module.
```suggestion
    let rc = block_on(mm_alice.rpc(&json!({
        "userpass": mm_alice.userpass,
        "method": "buy",
        "base": "MYCOIN",
        "rel": "MYCOIN1",
        "price": 25,
        "volume": "2",
    })))
```